src/JobDefinition.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+namespace Barracuda\JobRunner;
+
+/**
+ * Class JobDefinition
+ * This class has configuration relevant to running the job. E.g. when and for how long.
+ */
+class JobDefinition
+{
+	/**
+	 * @var bool
+	 */
+	private $enabled;
+
+	/**
+	 * @var null|string
+	 */
+	private $run_time;
+
+	/**
+	 * @var int
+	 */
+	private $interval;
+
+	/**
+	 * @var int
+	 */
+	private $max_run_time;
+
+	/**
+	 * @var \ReflectionClass The instance of ReflectionClass of the Job class
+	 */
+	private $reflection;
+
+	/**
+	 * @var int The unix timestamp of when the job last started.
+	 */
+	private $last_run_time_start;
+
+	/**
+	 * @var int The unix timestamp of when the job last finished.
+	 */
+	private $last_run_time_finish;
+
+	/**
+	 * JobDefinition constructor.
+	 * @param bool   $enabled      True if the job should run, false if not.
+	 * @param string $run_time     When the job should run in the format of 14:00 (2 pm every day).
+	 * @param int    $interval     Amount of seconds in between job runs. Default is every 6 hours.
+	 * @param int    $max_run_time The max amount of time the job should run for. Default is 2 days.
+	 */
+	public function __construct($enabled = true, $run_time = null, $interval = 21600, $max_run_time = 172800)
+	{
+		$this->enabled = $enabled;
+		$this->run_time = $run_time;
+		$this->interval = $interval;
```

Comment:
I think this object should be pretty dumb. I like validating that in addJob inside JobRunner